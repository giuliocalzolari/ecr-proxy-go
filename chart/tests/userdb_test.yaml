suite: test UserDB
templates:
- userdb.yaml
release:
  name: test-release
  namespace: test-namespace
kubernetesProvider:
  scheme:
    "v1/Secret":
      gvr:
        version: "v1"
        resource: "secrets"
      namespaced: true
tests:
- it: manifest should match snapshot
  chart:
    version: 1.2.3
  set:
    userAuth:
      enabled: true
  asserts:
  - matchSnapshot: {}

- it: should generated admin password
  documentSelector:
    path: metadata.name
    value: ecr-proxy-userdb-admin
  asserts:
  - equal:
      path: data.username
      value: admin
      decodeBase64: true
  - matchRegex:
      path: data.password
      pattern: ^[a-zA-Z0-9]{16}$
      decodeBase64: true

- it: should set admin password
  documentSelector:
    path: metadata.name
    value: ecr-proxy-userdb-admin
  set:
    userAuth:
      adminPassword: "test"
  asserts:
  - equal:
      path: data.password
      value: test
      decodeBase64: true


- it: should read admin password from secret
  kubernetesProvider:
    objects:
      - kind: Secret
        apiVersion: v1
        metadata:
          name: ecr-proxy-userdb-admin
          namespace: test-namespace
        data:
          password: dGVzdAo=
  documentSelector:
    path: metadata.name
    value: ecr-proxy-userdb-admin
  asserts:
  - equal:
      path: data.password
      value: test
      decodeBase64: true

