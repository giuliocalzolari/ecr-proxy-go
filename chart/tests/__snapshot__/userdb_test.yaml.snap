manifest should match snapshot:
  1: |
    apiVersion: v1
    data:
      userdb: W3sicGFzc3dvcmQiOiJnVUJrbE56Uno3bkFxYmdjIiwidXNlcm5hbWUiOiJhZG1pbiJ9XQ==
    kind: Secret
    metadata:
      labels:
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: ecr-proxy
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: ecr-proxy-1.2.3
      name: ecr-proxy-userdb
    type: opaque
  2: |
    apiVersion: v1
    data:
      password: Z1VCa2xOelJ6N25BcWJnYw==
      username: YWRtaW4=
    kind: Secret
    metadata:
      annotations:
        description: cred for admin
      labels:
        app.kubernetes.io/instance: test-release
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: ecr-proxy
        app.kubernetes.io/version: 0.0.0
        helm.sh/chart: ecr-proxy-1.2.3
      name: ecr-proxy-userdb-admin
